src/contribute/precommit-triage-guide.md

```diff
@@ -0,0 +1,125 @@
+permalink: /contribute/precommit-triage-guide/
+section_menu: section-menu/contribute.html
+---
+<!--
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+# Precommit Slowness Triage Guide
+
+Beam precommit jobs are suites of tests run automatically on Jenkins build
+machines for each pull request (PR) submitted to
+[apache/beam](https://github.com/apache/beam). For more information and the
+difference between precommits and postcommits, see
+[testing](/contribute/testing/).
+
+## What are fast precommits?
+
+Precommit tests are required to pass before a pull request (PR) may be merged.
+When these tests are slow they slow down Beam's development process.
+
+The aim is to have 95% of precommit jobs complete within 30 minutes
+(failing or passing).
+Technically, the 95th percentile of running time should be below 30 minutes over
+the past 4 weeks, where running time is the duration of time the job spends in
+the Jenkins queue + the actual time it spends running.
+
+## Detemining Slowness
+
+The current method for determining if precommmits are slow is to look at the
+[Jupyter
+notebook](https://github.com/apache/beam/tree/master/.test-infra/jupyter)
+`precommit_job_times.ipynb`.
+
+Run the notebook. It should output a table with running times. The numbers in
+the column `totalDurationMinutes_all` and the rows with a `job_name` like `4
+weeks 95th` contain the target numbers for determining slowness.
+If any of these numbers are above 30, triaging is required.
+
+### Example
+Here's an example table of running times:
+![example precommit duration table](/images/precommit_durations.png)
+
+In this example, Go precommits are taking approximately 14 minutes, which is
+fast. Java and Python precommits are taking 78 and 32 minutes respectively,
+which is slow. Both Java and Python precommits require triage.
+
+## Triage Process
+
+1. [Search for existing
+   issues](https://issues.apache.org/jira/issues/?filter=12344461)
+1. Create a new issue if needed: [Apache
+   JIRA](https://issues.apache.org/jira/issues)
+  - Project: Beam
+  - Components: testing, anything else relevant
+  - Label: precommit
+  - Reference this page in the description.
+1. Determine where the slowness is coming from and identify issues. Open
+   additional issues if needed (such as for multiple issues).
+1. Assign the issue as appropriate, e.g., to the test's or PR's author.
+
+## Resolution
+
+It is expected that slowness is resolved promptly. See [precommit test
```

Comment:
what about something like this for the first sentence:
It is important that we quickly fix slow precommit tests.